test/e2e/utils/generate_incident.go

```diff
@@ -0,0 +1,211 @@
+	"io"
+	"net/http"
+	"time"
+
+	sdk "github.com/PagerDuty/go-pagerduty"
+)
+
+const (
+	ClusterHasGoneMissing                         = "ClusterHasGoneMissing"
+	ClusterProvisioningDelay                      = "ClusterProvisioningDelay"
+	ClusterMonitoringErrorBudgetBurnSRE           = "ClusterMonitoringErrorBudgetBurnSRE"
+	InsightsOperatorDown                          = "InsightsOperatorDown"
+	MachineHealthCheckUnterminatedShortCircuitSRE = "MachineHealthCheckUnterminatedShortCircuitSRE"
+	ApiErrorBudgetBurn                            = "ApiErrorBudgetBurn"
+	AlertManagerDown                              = "AlertManagerDown"
+
+	// PagerDuty Events API endpoint
+	PagerDutyEventsURL = "https://events.pagerduty.com/v2/enqueue"
+)
+
+// EventPayload represents the payload structure for PagerDuty Events API
+type EventPayload struct {
+	Summary   string                 `json:"summary"`
+	Source    string                 `json:"source"`
+	Severity  string                 `json:"severity"`
+	Timestamp string                 `json:"timestamp"`
+	Details   map[string]interface{} `json:"custom_details,omitempty"`
+}
+
+// Event represents the complete event structure for PagerDuty Events API
+type Event struct {
+	RoutingKey  string        `json:"routing_key"`
+	EventAction string        `json:"event_action"`
+	DedupKey    string        `json:"dedup_key,omitempty"`
+	Payload     *EventPayload `json:"payload"`
+}
+
+// EventResponse represents the response from PagerDuty Events API
+type EventResponse struct {
+	Status   string `json:"status"`
+	Message  string `json:"message"`
+	DedupKey string `json:"dedup_key"`
+}
+
+func GetAlertSummary(alertName string) (string, error) {
+	switch alertName {
+	case ClusterHasGoneMissing:
+		return "cadtest has gone missing", nil
+	case ClusterProvisioningDelay:
+		return "ClusterProvisioningDelay -", nil
+	case ClusterMonitoringErrorBudgetBurnSRE:
+		return "ClusterMonitoringErrorBudgetBurnSRE Critical (1)", nil
+	case InsightsOperatorDown:
+		return "InsightsOperatorDown", nil
+	case MachineHealthCheckUnterminatedShortCircuitSRE:
+		return "MachineHealthCheckUnterminatedShortCircuitSRE CRITICAL (1)", nil
+	case ApiErrorBudgetBurn:
+		return "api-ErrorBudgetBurn k8sgpt test CRITICAL (1)", nil
+	case AlertManagerDown:
+		return "Alert Manager Down", nil
+	default:
+		return "", fmt.Errorf("unknown alert name: %s", alertName)
+	}
+}
+
+type TestPagerDutyClient interface {
+	CreateRequest(alertName, clusterID string) (string, error)
+	GetIncidentID(dedupKey string) (string, error)
+	ResolveIncident(incidentID string) error
+}
+
+type client struct {
+	routingKey string
+	apiClient  *sdk.Client
+	httpClient *http.Client
+}
+
+func NewClient(routingKey string) TestPagerDutyClient {
+	return &client{
+		routingKey: routingKey,
+		apiClient:  sdk.NewClient(routingKey),
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}
+}
+
+func (c *client) CreateRequest(alertName, clusterID string) (string, error) {
```

Comment:
Wouldn't it be much easier to use the `CreateEvent` sdk function? e.g.
https://github.com/PagerDuty/go-pagerduty/blob/9831333ebe6bbe82a890c9aa0a9fd462a2a8d3e8/examples/event.go#L15